step03/db.py

```diff
@@ -0,0 +1,68 @@
+from flask import g
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy import create_engine
+import models
+from models import Profiles
+
+Base = models.Base
+
+engine = create_engine('sqlite+pysqlite:///imagely.db')
+Base.metadata.create_all(engine)
+Session = sessionmaker(bind=engine)
+session = Session()
```

Comment:
I'm worried that this session object is not thread safe. If session only represents one connection to the database, your API will not be able to handle multiple requests at the same time safely because it's communicating all db queries through on stream.

I would fix the code in general so you can simply call `User.query.get(LOCAL_USER['username'])` (from step03B) to interact with the db. This creates sessions dynamically. Otherwise, if you want to manage session objects yourself, just make sure you make a new session object every time.

In reality, creating lots of connections to a db is expensive for the client side (API server) because it has to create new connections and can also tax the database CPU for the same reasons. There are also limits to the number of active connections that can be made. What you may need to do in the real world is to manage a pool of active connections (for example, 10) and dynamically grab an available connection object from the pool and simply hang when there are no free connections.